c/shared/source/uforead/uforead.c

```diff
@@ -1682,32 +1681,39 @@ static char* getXmlAttrValue(xmlAttr *attr){
     return (char*) attr->children->content;
 }
 
-static void parseXMLLib(ufoCtx h, xmlNodePtr cur) {
+static bool setXMLLib(ufoCtx h, xmlNodePtr cur, char* keyName) {
     char* keyValue;
 
+    if (h->parseState.UFOFile == preParsingGLIF) {     /* when called from preParseGLIF */
+        keyValue = parseXMLKeyValue(h, cur);
+        if (keyValueValid(h, cur, keyValue, keyName)) {
+            if (strEqual(keyName, "com.adobe.type.cid.CID")) {
+                h->parseState.GLIFInfo.currentCID = atoi(keyValue);
+                h->top.sup.flags |= ABF_CID_FONT;
+                h->top.sup.srcFontType = abfSrcFontTypeUFOCID;
+            } else if (strEqual(keyName, "com.adobe.type.cid.iFD"))
+                h->parseState.GLIFInfo.currentiFD = atoi(keyValue);
+        }
+    } else if (h->parseState.UFOFile == parsingGLIF) {  /* when called from parseGLIF */
+        if (strEqual(keyName, "com.adobe.type.autohint.v2")) {
+            h->parseState.type1HintDataV2 = true;
+            if (cur != NULL)
```

Comment:
If `cur` is NULL, do we want to just ignore the fact there was a recognized key but no value here? Seems like that should be a warning at least.

Reply:
For now, we've been aiming to reduce the amount of warnings until we add the verbosity flag option. However, I'm curious what @frankrolf thinks about this specifically. Frank, if within a UFO's .glif file's <lib> there is an empty `com.adobe.type.autohint.v2`, would it be helpful to the user to receive a warning?

Reply:
My experimental repo branch (not yet copied back from my machine) has already pulled all the C-code-level logging messages into a single framework with multiple "levels". So after we start working with that code it will be easier to unify the logging across all of our tools.

Reply:
Sounds good! So we can keep the warning messages for when we work with your changes